include/tvm/ir/module.h

```diff
@@ -57,6 +57,8 @@ class IRModuleNode : public Object {
   Map<GlobalVar, BaseFunc> functions;
   /*! \brief A map from global type vars to ADT type data. */
   Map<GlobalTypeVar, TypeData> type_definitions;
+  /*! \brief The external module containing the external functions used by this module. */
+  Array<runtime::Module> external_mods;
```

Comment:
Not sure if it's a right way to do so by fixing IR with Runtime

Reply:
@comaniac It would be great if we could have a discussion about this instead of you just saying it's not the correct approach. 

Currently, the external modules are stored in the IRModule attributes, so they are already mixed, it's just not made explicit in the AST. This change just moves the external modules out of the attributes and into the IRModule as a field.

Reply:
As @electriclilies says we already have this, and it is already essential to making IRModule the carrier for all passes, including the existing BYOC (primitive function to runtime Module) and the TIRToRuntime extension point from @Mousius .

Perhaps you are suggesting we need a way to represent 'already compiled' functions that's not bound up with the runtime?

Reply:
I don't think we are directly saying "no" here. To better organize our discussion:
1. It mixes compile-time IR and the runtime thing - usually it's done by a linker instead.
2. There exists better alternatives to use metadata module.

Cody mainly wanted to introduce some history in the design rationale.